drivers/serial/uart_stm32.c

```diff
@@ -824,7 +824,7 @@ static int uart_stm32_fifo_fill_visitor(const struct device *dev, const void *tx
 {
```

Comment:
The typedef `fifo_fill_fn` takes a `uint8_t` for third parameter, but that should also be changed by an `int`, that impacts the functions `fifo_fill_with_u8` and `fifo_fill_with_u16` as well.